src/main/java/org/junitpioneer/jupiter/AbstractEntryBasedExtension.java

```diff
@@ -147,7 +147,11 @@ public void afterAll(ExtensionContext context) throws Exception {
 	}
 
 	private void restoreOriginalEntries(ExtensionContext context) {
-		context.getStore(NAMESPACE).get(BACKUP, EntriesBackup.class).restoreBackup();
+		getStore(context).get(BACKUP, EntriesBackup.class).restoreBackup();
+	}
+
+	private Store getStore(ExtensionContext context) {
+		return context.getStore(Namespace.create(getClass()));
```

Comment:
Always creates a new `Namespace`?

Reply:
Consider the following solution:
```java
protected abstract Namespace namespace();
```
which in turn gets turned into
```java
private static final Namespace NAMESPACE = Namespace.create(...);

@Override
protected Namespace namespace() {
  return NAMESPACE;
}
```
Is that too complex?

Reply:
I borrowed this from JUnit's `TimingExtension` [example](https://junit.org/junit5/docs/current/user-guide/#extensions-lifecycle-callbacks-timing-extension). It may not be ideal, but I'm not sure if it's worth to further blow up the base class -- we already have 6 `abstract` methods.

Alternatively, we could also set it in the abstract ctor. Something like:

```java
abstract class AbstractEntryBasedExtension {

    private final Namespace namespace;

    protected AbstractEntryBasedExtension() {
        this.namespace = Namespace.create(getClass());
    }

}
```

What do you prefer? Keep it as it is, additional `abstract` method, abstract ctor, ...?

Reply:
It doesn't really matter, it's a very minor thing.

Reply:
Out of all the possible solution, yours is probably the easiest to understand (it's all in one place) and it shouldn't impact performance. Let's leave it as it is.